src/Console/CommandFactory.php

```diff
@@ -44,7 +44,7 @@ public function __construct(?ContainerInterface $container = null)
      */
     public function create(string $className): CommandInterface
     {
-        if ($this->container && $this->container->has($className)) {
+        if ($this->container instanceof \Cake\Core\ContainerInterface && $this->container->has($className)) {
```

Comment:
Are instanceof checks more efficient than implicit boolean conversion?

Reply:
> Are instanceof checks more efficient than implicit boolean conversion?

This change suggests it's not a container interface which doesn't seem right.

Reply:
> This change suggests it's not a container interface which doesn't seem right.

Yes, it's misleading. I would rather keep this unchanged.